cfme/storage/manager.py

```diff
@@ -0,0 +1,233 @@
+
+
+class ManagerDetailsView(ManagerView):
+    """The details page for Storage Manager or Provider"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+
+        return(
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(ManagerDetailsToolbar)
+    sidebar = View.nested(ManagerDetailsAccordion)
+    entities = View.nested(ManagerDetailsEntities)
+
+
+class ManagerTagsView(ManagerView):
+    """The tag page for Storage Manager or Provider"""
+    breadcrumb = BreadCrumb()
+    select_tag = BootstrapSelect('tag_cat')
+    select_value = BootstrapSelect('tag_add')
+    save = Button('Save')
+    reset = Button('Reset')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_manager and
+            self.breadcrumb.active_location == 'Tag Assignment')
+
+
+class ManagePoliciesView(ManagerView):
+    """The policies page for Storage Manager or Provider"""
+    breadcrumb = BreadCrumb()
+    policies = BootstrapTreeview("protectbox")
+    save_button = Button("Save")
+    reset_button = Button("Reset")
+    cancel_button = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_manager and
+            self.breadcrumb.active_location == "'Storage Manager' Policy Assignment")
+
+
+class BlockManagerCollection(BaseCollection):
+    """Collection object [block manager] for the :py:class:'cfme.storage.manager'"""
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+        self.nav = VersionPick({
+            Version.lowest(): ['Storage', 'Storage Providers'],
+            '5.8': ['Storage', 'Block Storage', 'Managers']})
+        self.manager_type = 'Block Storage Managers'
+
+    def instantiate(self, name, provider):
+        return BaseManager(self, name, provider)
+
+
+class ObjectManagerCollection(BaseCollection):
+    """Collection object [object manager] for the :py:class:'cfme.storage.manager'"""
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+        self.nav = VersionPick({
+            Version.lowest(): ['Storage', 'Storage Providers'],
+            '5.8': ['Storage', 'Object Storage', 'Managers']})
+        self.manager_type = 'Object Storage Managers'
+
+    def instantiate(self, name, provider):
+        return BaseManager(self, name, provider)
+
+
+class BaseManager(BaseEntity):
+    """ Model of an storage manager in cfme
+
+    Args:
+        collection: Instance of collection
+        name: Name of the object manager.
+        provider: Provider
+    """
+
+    def __init__(self, collection, name, provider):
+        self.collection = collection
+        self.appliance = self.collection.appliance
+        self.name = name
+        self.provider = provider
+        self.nav = self.collection.nav
```

Comment:
^^

Reply:
@izapolsk, I have to take nav [navigation_path] attribute from the collection. it's helping me to suppress "super method" with two different classes.

Reply:
I already wrote you about nav name. it is confusing and doesn't explain what it contains.